document-models/real-world-assets/src/_utils/generators.ts

```diff
@@ -26,3 +29,29 @@ export function makeFixedIncomeAssetWithDerivedFields(
 
     return newAsset;
 }
+
+export function makeRwaDocumentWithAssetCurrentValues(
+    document: RealWorldAssetsDocument,
+    currentDate = new Date(),
+) {
+    const documentCopy = copy(document);
```

Comment:
I've never tried it but [structuredClone]( https://developer.mozilla.org/en-US/docs/Web/API/structuredClone) might be good for these clone operations.

Reply:
I actually started off using structured clone for other similar things in this repo, but the problem with it is it doesn't work nicely with inferring types. Ends up needing a ton of annotations. This lib does the same job but with nice inference